vcd/resource_vcd_cse_kubernetes_cluster.go

```diff
@@ -0,0 +1,796 @@
+package vcd
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"github.com/hashicorp/go-cty/cty"
+	semver "github.com/hashicorp/go-version"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"sort"
+	"time"
+)
+
+func resourceVcdCseKubernetesCluster() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdCseKubernetesClusterCreate,
+		ReadContext:   resourceVcdCseKubernetesRead,
+		UpdateContext: resourceVcdCseKubernetesUpdate,
+		DeleteContext: resourceVcdCseKubernetesDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdCseKubernetesImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"cse_version": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"4.1.0", "4.1.1", "4.2.0"}, false),
+				Description:  "The CSE version to use",
+				DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+					// This custom diff function allows to correctly compare versions.
```

Comment:
Can you give a bit more details? Does it avoid `ForceNew` being triggered? Does is still show `diff` when changed?

Reply:
Yes, that would mean that if Go SDK returns, for any reason (currently I'd say that it's impossible) `CseVersion: 4.2`, and we had specified `cse_version = "4.2.0"` in the resource, with this function Terraform considers that `4.2==4.2.0` and does not trigger a Plan to change it.

I'd say this validation is not needed, as we work with SemVer in the SDK side, so I doubt this one is 100% needed. I left it as a double-check and also to make it clear that we work with SemVer objects.

But I can remove it.